pkg/cluster/check/etcd.go

```diff
@@ -0,0 +1,69 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package check
+
+import (
+	"context"
+	"fmt"
+
+	machineapi "github.com/talos-systems/talos/pkg/machinery/api/machine"
+	"github.com/talos-systems/talos/pkg/machinery/client"
+	"github.com/talos-systems/talos/pkg/machinery/config/types/v1alpha1/machine"
+)
+
+// EtcdConsistentAssertion checks that etcd membership is consistent across nodes.
+//nolint:gocyclo
+func EtcdConsistentAssertion(ctx context.Context, cluster ClusterInfo) error {
+	cli, err := cluster.Client()
+	if err != nil {
+		return err
+	}
+
+	nodes := append(cluster.NodesByType(machine.TypeInit), cluster.NodesByType(machine.TypeControlPlane)...)
+	nodesCtx := client.WithNodes(ctx, nodes...)
+	resp, err := cli.EtcdMemberList(nodesCtx, &machineapi.EtcdMemberListRequest{})
+	if err != nil {
+		return err
+	}
+
+	type data struct {
+		hostname  string
+		id        uint64
+		isLearner bool
+	}
+
+	knownMembers := map[data]struct{}{}
+
+	for i, message := range resp.GetMessages() {
```

Comment:
it makes sense to sort `Messages` based on the `Metadata.Hostname`, so that error reported is consistent, otherwise it prints different error each poll, as the first message is different each time

Reply:
Done!